server/lomas_server/dp_queries/dp_libraries/opendp.py

```diff
@@ -81,28 +83,128 @@ def get_lf_domain(metadata):
 
     # Margins
     # TODO Check lengths vs. keys for public info -> not in doc anymore.
+
+    # Global params
+    params = {}
     if "rows" in metadata:
+        params["max_partition_length"] = metadata["rows"]
+    if "max_ids" in metadata:
+        params["max_num_partitions"] = metadata["max_ids"]
+    if len(params) > 0:
         lf_domain = dp.domains.with_margin(
             lf_domain,
             by=[],
             public_info="lengths",
-            max_partition_length=metadata["rows"],
-            max_num_partitions=1,
+            **params,
         )
 
-    for name, series_info in metadata["columns"].items():
-        # TODO add max_partition_contributions, max_influenced_partitions
-        if "max_num_partitions" in series_info:
-            lf_domain = dp.domains.with_margin(
-                lf_domain,
-                by=[name],
-                public_info="lengths",
-                max_num_partitions=series_info["max_num_partitions"],
-            )
+    # Grouping logic (margin adaptation)
```

Comment:
It is great that we already have a solutino! I'll try to have a look too later to see if there is a miraculous way